### PR TITLE
Support ruby http gem v1.0.0+

### PR DESCRIPTION
**Pull request by ixti**

### CODE REVIEW

File: lib/webmock/http_lib_adapters/http_rb/response.rb [hunk 1]
```diff
@@ -14,9 +14,17 @@ def self.from_webmock(webmock_response, request_signature = nil)
       status  = Status.new(webmock_response.status.first)
       headers = webmock_response.headers || {}
       body    = Body.new Streamer.new webmock_response.body
-      uri     = URI request_signature.uri.to_s if request_signature
+      uri     = URI.coerce request_signature.uri.to_s if request_signature
```
Comment (nerdrew) [inline lib/webmock/http_lib_adapters/http_rb/response.rb]: This is failing for me.

```
     NoMethodError:
       undefined method `coerce' for HTTP::URI:Class
```

Also fails if I use `::URI.coerce`

```
     NoMethodError:
       undefined method `coerce' for URI:Module
```

Comment (ixti) [inline lib/webmock/http_lib_adapters/http_rb/response.rb]: Oh. I don't know what I was thinking of when pushed that. Thanks for catching this up.

**Comment by ixti** [comment]
Do not merge yet - seems like Addressable::URI changes some of the internal stuff - gonna check and will get back to this ASAP

**Comment by ixti** [comment]
Issue was not caused by Addressable::URI, but by WebMock appending default port to URI within request signature.

**Comment by ajb** [comment]
:+1: on this change -- running into this issue myself and took me a second to debug :smile:

**Comment by ixti** [comment]
@davidbegin should be fine now :D

**Comment by davidbegin** [comment]
Thanks! I'll take a look today.

**Comment by davidbegin** [comment]
:cake: Thanks so much for fixing this! :cake: 

I rebased it on master, relaxed the http rb version, and merged it in over at #559. 
So I am going to close this.

**Comment by ixti** [comment]
Awesome! Thanks!

**Pull request closed without merging**